bats/tests/k8s/up-downgrade-k8s.bats

```diff
@@ -0,0 +1,140 @@
+# Test case 8 & 13
+
+# shellcheck disable=SC2030,SC2031
+# SC2030 (info): Modification of output is local (to subshell caused by @bats test).
+# SC2031 (info): output was modified in a subshell. That change might be lost.
+
+setup() {
+    load '../helpers/load'
+}
+
+@test 'factory reset' {
+    factory_reset
+}
+
+@test 'start rancher desktop' {
+    start_kubernetes
+    wait_for_apiserver
+    # the docker context "rancher-desktop" may not have been written
+    # even though the apiserver is already running
+    wait_for_container_engine
+}
+
+@test 'deploy nginx - always restart' {
+    ctrctl pull nginx
+    run ctrctl run -d -p 8585:80 --restart=always --name nginx-restart nginx
+    assert_success
+}
+
+@test 'deploy nginx - no restart' {
+    run ctrctl run -d -p 8686:80 --restart=no --name nginx-no-restart nginx
+    assert_success
+}
+
+@test 'deploy busybox' {
+    run kubectl_exe create deploy busybox --image=busybox --replicas=2 -- /bin/sh -c "sleep inf"
+    assert_success
+}
+
+verify_nginx() {
+    local nginx_ports=(8585 8686)
+    for port in "${nginx_ports[@]}"; do
+        run curl http://localhost:"$port"
```

Comment:
There is no reason to quote `$port` because we know it is just a number. But if you still want to quote all interpolation. I find quoting the whole string easier to read:

```suggestion
        run curl "http://localhost:$port"
```